app/models/host/managed.rb

```diff
@@ -151,6 +151,8 @@ class Jail < ::Safemode::Jail
 
   scope :for_vm, ->(cr,vm) { where(:compute_resource_id => cr.id, :uuid => Array.wrap(vm).compact.map(&:identity).map(&:to_s)) }
 
+  scope :with_compute_resource, -> { where('uuid is not null AND compute_resource_id is not null') }
```

Comment:
Isn't `where.not(:compute_resource_id => nil)` also equivalent? I emphasize the use of AR constructs vs SQL just to leave SQL optimizations up to the ORM & possible security holes